Client/mods/deathmatch/logic/CClientPed.cpp

```diff
@@ -4626,7 +4639,7 @@ bool CClientPed::IsOnGround()
     CVector vecPosition;
     GetPosition(vecPosition);
     float fGroundLevel = static_cast<float>(g_pGame->GetWorld()->FindGroundZFor3DPosition(&vecPosition));
-    return (vecPosition.fZ > fGroundLevel && (vecPosition.fZ - fGroundLevel) <= 1.0f);
+    return (vecPosition.fZ > fGroundLevel && (vecPosition.fZ - fGroundLevel) <= 1.01f);
```

Comment:
```suggestion

    // We don't care if the ped is underground
    if ((vecPosition.fZ - fGroundLevel) <= FLOAT_EPSILON)
        return false;
        
    // Check that (zPos - groundLevel) <= 1, but also account for epsilon
    return (vecPosition.fZ - fGroundLevel - 1.0f) <= FLOAT_EPSILON;
```

Does this do the same thing?

Reply:
let me test

Reply:
since FLOAT_EPSILON expands to 0.0001f it should work, gonna test it rn.

Reply:
My suggested code seems theoretically incorrect (even if it might happen to work).

Looking at https://stackoverflow.com/a/77735/1517394:

> It all depends on context.
>
> I have spent a long time tracing a bugs in a system that presumed `a==b` if `|a-b|<epsilon`. The underlying problems were: [...]

And https://stackoverflow.com/a/253874/1517394:

> The following definitions are from [The art of computer programming by Knuth](http://books.google.ca/books?ei=KRkLSYLZEJ7ktQOX3-ChBQ&id=T89QAAAAMAAJ&dq=%22essentially+equal%22+%22approximately+equal%22+knuth&q=%22essentially+equal%22+%22approximately+equal%22+%22definitely+less+than%22&pgis=1#search):
> 
> ```cpp
> bool approximatelyEqual(float a, float b, float epsilon)
> {
>     return fabs(a - b) <= ( (fabs(a) < fabs(b) ? fabs(b) : fabs(a)) * epsilon);
> }
> 
> bool essentiallyEqual(float a, float b, float epsilon)
> {
>     return fabs(a - b) <= ( (fabs(a) > fabs(b) ? fabs(b) : fabs(a)) * epsilon);
> }
> 
> bool definitelyGreaterThan(float a, float b, float epsilon)
> {
>     return (a - b) > ( (fabs(a) < fabs(b) ? fabs(b) : fabs(a)) * epsilon);
> }
> 
> bool definitelyLessThan(float a, float b, float epsilon)
> {
>     return (b - a) > ( (fabs(a) < fabs(b) ? fabs(b) : fabs(a)) * epsilon);
> }
> ```

Reply:
Yeah probably, well we can lower the "1.01f" more according to the values that I got above, "1.01f" is already alot

Reply:
Why not use 1.001 instead of 1.01? 0.001 is very very low and no one will notice an issue with this, pretty sure

Reply:
yeah I thought about it, just waiting for lpsd to test it